metadata-ingestion/src/datahub/ingestion/api/auto_work_units/auto_ensure_aspect_size.py

```diff
@@ -0,0 +1,80 @@
+import json
+import logging
+from typing import Iterable, List
+
+from datahub.emitter.rest_emitter import INGEST_MAX_PAYLOAD_BYTES
+from datahub.emitter.serialization_helper import pre_json_transform
+from datahub.ingestion.api.workunit import MetadataWorkUnit
+from datahub.metadata.schema_classes import (
+    DatasetProfileClass,
+    SchemaFieldClass,
+    SchemaMetadataClass,
+)
+
+logger = logging.getLogger(__name__)
+
+
+def ensure_dataset_profile_size(profile: DatasetProfileClass) -> None:
+    """
+    This is quite arbitrary approach to ensuring dataset profile aspect does not exceed allowed size, might be adjusted
+    in the future
+    """
+    sample_fields_size = 0
+    if profile.fieldProfiles:
+        logger.debug(f"Length of field profiles: {len(profile.fieldProfiles)}")
+        for field in profile.fieldProfiles:
+            if field.sampleValues:
+                values_len = 0
+                for value in field.sampleValues:
+                    if value:
+                        values_len += len(value)
+                logger.debug(
+                    f"Field {field.fieldPath} has {len(field.sampleValues)} sample values, taking total bytes {values_len}"
+                )
+                if sample_fields_size + values_len > INGEST_MAX_PAYLOAD_BYTES:
+                    logger.warning(
```

Comment:
ideally we'd pass the source reporter, and then we'd do `report.warning(...)` and this would show up as a warning in the final ingestion report in the UI

Reply:
@hsheth2 adjusted the code to use source report